js/classes/animator.js

```diff
@@ -44,28 +46,30 @@ class Animator {
             this.elapsedTime += GAME.clockTick;
         }
         
-        if (this.elapsedTime > this.totalTime && this.looped) this.elapsedTime -= this.totalTime;
-        const frame = this.currentFrame() + ((this.reversed) ? 1 : 0);
+        if (this.elapsedTime > this.totalTime) {
+            // The animation has completed
+            if (this.looped) {
+                this.elapsedTime -= this.totalTime;
+                this.currentFrame = 0; // Reset current frame at the start of the loop
+            } else {
+                this.currentFrame = this.frameCount - 1; // Set current frame to the last frame if not looping
```

Comment:
What is the point of this?

Reply:
I don't think I wrote that. Maybe I was using outdated code and that chunk of code was written by Trae or Devin a while ago.

Reply:
Weird! I didn't make this change. I was going to comment the same thing as Caleb because it looks like the new code does essentially the same thing.